src/mlpack/methods/time_series_forecast/exponential_smoothing/ses.cpp

```diff
@@ -0,0 +1,90 @@
+/**
+ * @file methods/time_series/exponential_smoothing/ses.cpp
+ * @author Aditi Pandey
+ *
+ * Implementation of Single exponential smoothing.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#include "ses.hpp"
+#include <mlpack/core/util/log.hpp>
+
+using namespace mlpack;
+using namespace mlpack::single_exponential_smoothing;
+
+
+double calc_mae(std::vector<double>& errors) {
+    double mae = 0.0;
+    for(std::vector<double>::iterator it = errors.begin(); it != errors.end(); it++) {
+        mae += std::abs(*it);
+    }
+    return mae/errors.size();
+}
+
+double calc_mse(std::vector<double>& errors) {
+    double mse = 0.0;
+    for(std::vector<double>::iterator it = errors.begin(); it != errors.end(); it++) {
+        mse += pow(*it, 2);
+    }
+    return mse/errors.size();
+}
+
+double calc_mape(std::vector<double>& data, std::vector<double>& errors) {
+    double mape = 0.0;
+    for(std::vector<double>::iterator it1 = errors.begin(), it2 = data.begin();\
+    it1 != errors.end(), it2 != data.end(); it1++, it2++) {
+        mape += std::abs(*it1 / *it2);
+    }
+    return mape/errors.size();
+}
+
+void SingleES::update(const double& data, const double& alpha) {
+    forecast_past.push_back(alpha*data + (1-alpha)*forecast_past.back());
+}
+
+void SingleES::SingleES(std::vector<double>& data, const double& alpha) {
+    this->alpha = alpha;
+    forecast_past.push_back(data.front());
```

Comment:
Not sure, maybe there is a reason to use `std::vector` instead of `arma::vec` or `arma::mat`. But ideally we stick with armadillo datatypes, which is what we use across the codebase.

Reply:
Yes surely, I'll change that to armadillo.

Reply:
Great, looking forward to see the changes.